pkgs/tools/typesetting/typst/default.nix

```
@@ -0,0 +1,53 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, stdenv
+, darwin
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "typst";
+  version = "unstable-2023-03-21";
+
+  src = fetchFromGitHub {
+    owner = "typst";
+    repo = "typst";
+    rev = "9fc432a41d9970dfff360fec225a926b5ab8f2ab";
+    hash = "sha256-fHKBPmq0YNNElnT6eM8zFFhrvRFReqcuqF8B5tCzaso=";
+  };
```

There are currently releases which might be more suitable than taking the latest commit.

```suggestion
  version = "22-03-21-2";

  src = fetchFromGitHub {
    owner = "typst";
    repo = "typst";
    rev = version;
    hash = "sha256-bJPGs/Bd9nKYDrCCaFT+20+1wTN4YdkV8bGrtOCR4tM=";
  };
```

---

didn't realize they were releases, fixed it